hello-vulkan/app/src/main/cpp/vk_main.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <android/asset_manager_jni.h>
+#include <android/native_window_jni.h>
+#include <android_native_app_glue.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <iostream>
+
+#include "vk_triangle/hellovk.h"
+
+/*
+ * Shared state for the app. This will be accessed within lifecycle callbacks
+ * such as APP_CMD_START or APP_CMD_INIT_WINDOW.
+ *
+ * We store:
+ * struct android_app - a pointer to the Android application handle
+ *
+ * vkt::HelloVK - a pointer to our (this) Vulkan application in order to call
+ *  the rendering logic
+ *
+ * bool canRender - a flag which signals that we are ready to call the vulkan
+ * rendering logic
+ *
+ */
+struct Engine {
+  struct android_app *app;
+  vkt::HelloVK *app_backend;
+  bool canRender = false;
```

Comment:
let's advocate Vulkan here?  name it VulkanEngine? (hahaha, but useful, I think).